rllib/agents/dqn/simple_q.py

```diff
@@ -215,3 +250,82 @@ def execution_plan(workers, config, **kwargs):
         )
 
         return StandardMetricsReporting(train_op, workers, config)
+
+    @ExperimentalAPI
+    def training_iteration(self) -> ResultDict:
+        """Simple Q training iteration function.
+
+        This replicates Simple Q's execution_plan behaviour, e.g.:
+        - For every time we (1) sample (MultiAgentBatch) from workers...
+            - (2) Sample training batch (MultiAgentBatch) from replay buffer.
+            - (3) Learn on training batch.
+            - (4) Update target network every target_network_update_freq steps.
+        - (5) Concatenate freshly collected samples.
+        - (6) Store new samples in replay buffer.
+        - (7) Return all collected metrics for the iteration.
+
+        Returns:
+            The results dict from executing the training iteration.
+        """
+        batch_size = self.config["train_batch_size"]
+        local_worker = self.workers.local_worker()
+
+        # Collects SampleBatches in parallel and synchronously
+        # from the Trainer's RolloutWorkers until we hit the
+        # configured `train_batch_size`.
+        sample_batches = []
+        num_env_steps = 0
+        num_agent_steps = 0
+        train_results = {}
+
+        while (not self._by_agent_steps and num_env_steps < batch_size) or (
```

Comment:
I think this is incorrect here.
The train_batch size should only be used for sampling from the buffer, not for the rollouts.
Rollouts should simply be one-per-worker-per-iteration for SimpleQ (plus add all of them to replay buffer).

Reply:
@sven1977 is right in this case. 

Is getting a large enough batch from `self.local_replay_buffer.sample(batch_size)` not enough? If so, you just need to verify that train_batch_size number of timesteps is being trained on every time we sample from the replay buffer, and then after that, if the number of agent steps trained does not match with the execution plan API implementation, then increase the train batch size in your implementation.

Reply:
Thanks for the input. The sample batch size of the replay buffer is consistent. Only that for every time we sample a fragment from a rollout worker, we train for one iteration - in the execution plan and now in the training iteration fn. Changing this breaks learning atm.